mabl snippets/GetDateComponents.js

```diff
@@ -118,110 +164,82 @@ Date.prototype.getPreviousWeekday = function (weekday) {
   return date;
 };
 
-// Formats the date based off of the provided options
-// @param {object {string: string}} option - The options for how to format the date string
-// @return {string} - Formatted date matching options
-// Formatting the date components:
-// Formatting Options: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat
-Date.prototype.format = function (option) {
-  var date = new Date(this.valueOf());
-  return date.toLocaleDateString("en-US", option);
-};
-
-
 /**************** Date Formatting Methods ****************/
 
-
 // Adds a method to Date objects that gets the month matching a format
 Date.prototype.month = function (format) {
-  var region =
-    arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "en-us";
-  return this.toLocaleDateString(region, {
+  let options = {
     month: format,
-  });
+  };
+  if (timezone) {options.timeZone = timezone};
+  return this.toLocaleDateString(locale, options);
 };
 
 // Adds a method to Date objects that gets the day matching a format
 Date.prototype.day = function (format) {
-  var region =
-    arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "en-us";
-  return this.toLocaleDateString(region, {
+  let options = {
     day: format,
-  });
+  };
+  if (timezone) {options.timeZone = timezone};
+  return this.toLocaleDateString(locale, options);
 };
 
 // Adds a method to Date objects that gets the year matching a format
 Date.prototype.year = function (format) {
-  var region =
-    arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "en-us";
-  return this.toLocaleDateString(region, {
-    year: format,
-  });
+  let options = { year: format };
+  if (timezone) {options.timeZone = timezone};
+  return this.toLocaleDateString(locale, options);
 };
 
 // Adds a method to Date objects that gets the hour matching a format
-Date.prototype.hour = function (format) {
-  var hour12 =
-    arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
-  var region =
-    arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : "en-us";
-  var hours = this.toLocaleTimeString(region, {
+Date.prototype.hour = function (format, hour12) {
+  let options = {
     hour: format,
     hour12: hour12,
-  });
+  };
+  if (timezone) {options.timeZone = timezone};
+  var hours = this.toLocaleTimeString(locale, options);
   return hours.split(" ")[0];
 };
 
 // Adds a method to Date objects that gets the dayPeriod (AM/PM)
 Date.prototype.dayPeriod = function () {
-  var region =
-    arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "en-us";
-  var hours = this.toLocaleTimeString(region, {
+  let options = {
```

Comment:
I wonder if there could be a namespace collision with a customer JavaScript if they were to use the same names. I am wondering if we should a prefix like `mablDayPeriod` to avoid anything like that. I may be overthinking this.

Reply:
That variable should be scoped to that function in theory right?
What would you replace with "mablDayPeriod"? "options"?